agents/timestream_aggregator/timestream_aggregator.py

```diff
@@ -243,6 +248,18 @@ def read_frames(self):
 
         self.frames = self.reader.Process(None)
         if self.frames:
+            # Handle flow control frames
+            for f in self.frames:
+                if f.type in [core.G3FrameType.none]:
+                    # START; create_new_file
+                    if f.get('sostream_flowcontrol') == 1:
+                        self.close_file()
+                        self.create_new_file()
+
+                    # END; close_file
+                    if f.get('sostream_flowcontrol') == 2:
+                        self.close_file()
+
             # Discard flow control frames
             self.frames = [x for x in self.frames if x.type != core.G3FrameType.none]
```

Comment:
So all "none"-type frames are flow control?  Then we should print an error or warning if a badly formatted "none" frame is encountered (I.e. one with no sostream_flowcontrol information).

Reply:
Yeah, I believe right now all "none" type frames are for flow control. I added a warning to the logs if one doesn't contain the flowcontrol key.